parser-typechecker/src/Unison/Util/Star3.hs

```diff
@@ -160,13 +160,19 @@ insertD23 (f, x, y) s = Star3 fact' (d1 s) d2' d3' where
   d2'   = R.insert f x (d2 s)
   d3'   = R.insert f y (d3 s)
 
-deleteD23 :: (Ord fact, Ord d1, Ord d2, Ord d3)
-          => (fact, d2, d3)
+deleteD3 :: (Ord fact, Ord d1, Ord d2, Ord d3)
+          => (fact, d3)
+          -> Star3 fact d1 d2 d3
+          -> Star3 fact d1 d2 d3
+deleteD3 (f, x) s = Star3 (fact s) (d1 s) (d2 s) d3' where
+  d3' = R.delete f x (d3 s)
+
+deleteD2 :: (Ord fact, Ord d1, Ord d2, Ord d3)
+          => (fact, d2)
           -> Star3 fact d1 d2 d3
           -> Star3 fact d1 d2 d3
-deleteD23 (f, x, y) s = Star3 (fact s) (d1 s) d2' d3' where
+deleteD2 (f, x) s = Star3 (fact s) (d1 s) d2' (d3 s) where
```

Comment:
GitHub diff is misleading - I deleted the unused `deleteD23` and introduced `deleteD2` and `deleteD3`.